app/assets/javascripts/Components/Modals/autotest_specs_upload_modal.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import Modal from 'react-modal';
+
+class AutotestSpecsUploadModal extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      newFile: null
+    };
+  }
+
+  componentDidMount() {
+    Modal.setAppElement('body');
+  }
+
+  onSubmit = (event) => {
+    event.preventDefault();
+    if (!!this.state.newFile) {
+      this.props.onSubmit(this.state.newFile);
+      this.setState({newFile: null})
+    } else {
+      this.props.onRequestClose()
+    }
+  };
+
+  handleFileUpload = (event) => {
+    this.setState({newFile: event.target.files[0]})
```

Comment:
I could be wrong, but I think it's possible for `onChange` to be triggered if you already have a file selected, but then go to select another file and press "Cancel", which results in no file being selected.

So I'd add an emptiness check on `event.target.files` and if it's empty, update `newFile` to be `null`.